common/SC_ServerBootDelayWarning.cpp

```diff
@@ -0,0 +1,47 @@
+// Add a warning when Windows Defender delays scsynth boot by 60+ seconds
+// cf. github issue #4368
+#include "SC_ServerBootDelayWarning.h"
+
+#ifdef _WIN32
+#    include <iostream>
+#    include <thread>
+#    include <mutex>
+#    include <chrono>
+#    include <condition_variable>
+
+static bool g_isServerBooted = false;
+static std::mutex g_scsynthBootWarningMutex;
+static std::condition_variable g_scsynthBootWarningConditionVariable;
+static std::thread* g_bootWarningThread;
```

Comment:
this can be a `std::thread` -- the default constructor (number 1 [here](https://en.cppreference.com/w/cpp/thread/thread/thread)) doesn't start a real thread of execution. then you can assign to it with `g_bootWarningThread = std::thread(...)`

Reply:
in modern C++ (C++11 and newer), using `new` and `delete` directly is almost always a bad sign. there are several ways of avoiding it -- unique_ptr and shared_ptr among them -- that are not only semantically clearer, but also safer.

Reply:
Thanks ! I had not read the section about "utilities" in "A tour of C++" yet. A lot of things have changed (in a good way) since my old days in C++ :)